### ISSUE TITLE
SA1211 reported for incorrect comparison

### ISSUE THREAD

**Issue by sharwell** [DotNetAnalyzers/StyleCopAnalyzers#612]
Currently SA1211 is reported for the first of the following using alias directives:

```csharp
using StringBuilder = System.Text.StringBuilder;
using StringWriter = System.IO.StringWriter;
```

It appears the analysis is examining the target name, not the alias name.

**Comment by sharwell** [comment]
On further examination, that might not be the cause. I'm seeing the warning reported for `MemoryStream` in the following:

```csharp
using System;
using System.Collections.Generic;
using System.Net;
using System.Text;
using System.Threading;
using System.Threading.Tasks;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using Rackspace.Threading;
using MemoryStream = System.IO.MemoryStream;
using Stream = System.IO.Stream;
```

**Comment by AArnott** [comment]
Here is another failure case:

    using System;
    using System.Collections.Concurrent;
    using System.Collections.Generic;
    using System.Collections.Specialized;
    using System.Diagnostics;
    using System.Linq;
    using System.Reflection;
    using System.Runtime.CompilerServices;
    using System.Threading;
    using System.Threading.Tasks;
    using JoinableTaskSynchronizationContext = Microsoft.VisualStudio.Threading.JoinableTask.JoinableTaskSynchronizationContext;
    using SingleExecuteProtector = Microsoft.VisualStudio.Threading.JoinableTaskFactory.SingleExecuteProtector;
